tests/platform_tests/api/test_psu.py

```
@@ -200,3 +200,33 @@ def test_led(self, duthost, localhost, platform_api_conn):
                         self.expect(color == color_actual, "Status LED color incorrect (expected: {}, actual: {}) from PSU {}".format(color, color_actual, psu_id))
         self.assert_expectations()
 
+    def test_get_position_in_parent(self, platform_api_conn):
+        for psu_id in range(self.num_psus):
+            position = psu.get_position_in_parent(platform_api_conn, psu_id)
+            self.expect(position is not None, "Failed to perform get_position_in_parent for psu id {}".format(psu_id))
+            self.expect(isinstance(position, int), "Position value must be an integer value for psu id {}".format(psu_id))
+        self.assert_expectations()
+
+    def test_is_replaceable(self, platform_api_conn):
+        for psu_id in range(self.num_psus):
+            replaceable = psu.is_replaceable(platform_api_conn, psu_id)
+            self.expect(replaceable is not None, "Failed to perform is_replaceable for psu id {}".format(psu_id))
+            self.expect(isinstance(replaceable, bool), "Replaceable value must be a bool value for psu id [}".format(psu_id))
+        self.assert_expectations()
+
+    def test_thermals(self, platform_api_conn):
+        for psu_id in range(self.num_psus):
+            try:
+                num_thermals = int(psu.get_num_thermals(platform_api_conn, psu_id))
+            except:
```

To catch all exceptions, please use `except Exception:`